tests/tools.vitruv.applications.pcmumlclass.tests/src/tools/vitruv/applications/pcmumlclass/tests/PcmQueryUtil.xtend

```diff
@@ -0,0 +1,34 @@
+package tools.vitruv.applications.pcmumlclass.tests
+
+import edu.kit.ipd.sdq.activextendannotations.Utility
+import tools.vitruv.framework.views.View
+import org.palladiosimulator.pcm.repository.Repository
+import org.palladiosimulator.pcm.system.System
+
+import static org.junit.jupiter.api.Assertions.assertEquals
+
+@Utility
+class PcmQueryUtil {
+
+	static def getPcmSystem(View view) {
+		view.getRootObjects(System)
+	}
+
+	static def getPcmRepository(View view) {
+		view.getRootObjects(Repository)
+	}
+
+	static def claimPcmRepository(View view, String packageName) {
+		// primitiveDataTypes repository is not desired as output
+		assertEquals(
+			getPcmRepository(view).iterator.findFirst[it.entityName.equals(packageName)],
+			getPcmRepository(view).iterator.findLast[it.entityName.equals(packageName)],
+			"More than one repository was created."
+		)
+		getPcmRepository(view).iterator.findFirst[it.entityName.equals(packageName)]
```

Comment:
This can probably be written in a more compact way by using `claimOne()` from the [SDQ-Commons](https://github.com/kit-sdq/SDQ-Commons) class [`IterableUtil`](https://github.com/kit-sdq/SDQ-Commons/blob/fb3ef14b861294823ad1e3000affb6c44c7c049e/bundles/edu.kit.ipd.sdq.commons.util.java/src/edu/kit/ipd/sdq/commons/util/java/lang/IterableUtil.xtend).

Reply:
As far as I see claimOne() returns the only element in an Iterable without further restrictions for the element.
In this case, getPcmRepository returns two repositories, the base repository and primitiveDatatypes. AssertEquals checks that there is exactly one base repository. Should I change getPcmRepository to only return the base repository?

Reply:
You could simply do the filtering before calling claimOne like `getPcmRepository(view).iterator.filter[it.entityName.equals(packageName)].claimOne`